pydbus/xml_generator.py

```diff
@@ -0,0 +1,107 @@
+"""Automatic XML documentation generator."""
+import inspect
+import sys
+
+from xml.etree import ElementTree
+
+
+PROPERTY_EMITS_SIGNAL = "org.freedesktop.DBus.Property.EmitsChangedSignal"
+
+
+# Python 2 treats them as methods, Python 3 as functions
+ismethod = inspect.ismethod if sys.version_info[0] == 2 else inspect.isfunction
+
+
+def verify_arguments(function):
+	"""Verify that the function is correctly defined."""
+	args, vargs, kwargs, defaults = inspect.getargspec(function)
+	if defaults is not None:
+		raise ValueError(
+			"Default values are not allowed for method "
+			"'{}'".format(function.__name__))
+	# TODO: Check if vargs or kwargs are actually allowed in dbus.
```

Comment:
No, they aren't.

However, you should skip arguments with names starting from "dbus_", as "dbus_context" might be passed as a keyword argument automatically by pydbus: https://github.com/LEW21/pydbus/commit/78daafaf1f4e22962a71d2178774ed41a5b0274f